src/PhoneValidation.tsx

```diff
@@ -0,0 +1,73 @@
+import type { ChangeEvent } from 'react'
+import { useRef, useState } from 'react'
+import BackButton from './components/BackButton'
+import ValidationMessage from './components/ValidationMessage/ValidationMessage'
+// import CorrectValidation from './components/ValidationMessage'
+import './PhoneValidation.scss'
+
+type Props = {}
+
+const PhoneValidation = ({}: Props) => {
+  const inputsRef = [null, null, null, null].map(() => useRef<HTMLInputElement>(null))
+  const [numbers, setNumbers] = useState<number[]>([-1, -1, -1, -1])
+  const buttonRef = useRef<HTMLButtonElement>(null)
+  const [validated, setValidated] = useState<boolean>()
+
+  const OnChangeNumber = (event: ChangeEvent<HTMLInputElement>, index: number) => {
+    if (!event.target.value) return setNumbers((array) => array.map((value, i) => (i === index ? -1 : value)))
+    if (index < 3) inputsRef[index + 1]?.current?.focus()
+    else buttonRef?.current?.focus()
+
+    const new_number = Number(event.target.value[event.target.value.length - 1])
+    setNumbers((array) => array.map((value, i) => (i === index ? new_number : value)))
```

Comment:
```suggestion
    const newNumber = Number(event.target.value)
    const newNumbersState = numbers.map((value, i) => (i === index ? new_number : value))) 
    setNumbers(newNumberState)
```

Some suggestions here: 

1. Variables should be on camelCase
2. I don't understand this piece of code: `event.target.value[event.target.value.length - 1]`. Could you just access the value directly?
3. (Optional) I would do the map outside and store the value on a variable. It may keep the code clean. However, there is not much complexity, so you can leave it there